src/snowflake/connector/auth.py

```diff
@@ -237,13 +276,28 @@ def authenticate(
             {k: v for (k, v) in body["data"].items() if k != "PASSWORD"},
         )
 
+        # accommodate any authenticator specific timeout requirements here.
+        # login_timeout comes from user configuration.
+        # Between login timeout and auth specific
+        # timeout use whichever value is smaller
+        if getattr(auth_instance, "get_timeout", None) is not None:
```

Comment:
Since the actual value never gets used
```suggestion
        if hasattr(auth_instance, "get_timeout"):
```